mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreRightManager.java

```diff
@@ -124,6 +125,9 @@ public void applyRightsCommand(MailboxPath mailboxPath, ACLCommand mailboxACLCom
         MailboxMapper mapper = mailboxSessionMapperFactory.getMailboxMapper(session);
         Mailbox mailbox = mapper.findMailboxByPath(mailboxPath);
         mapper.execute(Mapper.toTransaction(() -> mapper.updateACL(mailbox, mailboxACLCommand)));
+
+        ACLDiff aclDiff = ACLDiff.computeDiff(mailbox.getACL(), mailbox.getACL().apply(mailboxACLCommand));
```

Comment:
a command and a diff should be exactly the same thing, there should exist a bijection between these objects, we should not need to apply the command.

Reply:
Yes and we have another commit to fix it up: https://github.com/linagora/james-project/pull/1121/commits/afd1a61193157a7e514fd968a5d9f1d4c7c6cfe3 
Then the mapper.updateACL return aclDiff directly, is that your comment?

Reply:
In replace mode, you can't have the diff from the command easily so it's not that easy, I was wrong.